src/stm32f1/pwm.cpp

```diff
@@ -0,0 +1,230 @@
+#include "pin.hpp"
+#include "power.hpp"
+#include "pwm_reg.hpp"
+#include <cmath>
+#include <libhal-arm-mcu/stm32f1/clock.hpp>
+#include <libhal-arm-mcu/stm32f1/constants.hpp>
+#include <libhal-arm-mcu/stm32f1/pin.hpp>
+#include <libhal-arm-mcu/stm32f1/pwm.hpp>
+#include <libhal-util/bit.hpp>
```

Comment:
Follow the https://libhal.github.io/4.1/contributor_guide/style/#s14-headers guidelines for header order.